### ISSUE TITLE
small update for network.h (softmax function in combination cross_entropy_multiclass)

### ISSUE THREAD

**Issue by zeno40** [nyanp/tiny-cnn#137]
void bprop_2nd(const vec_t& out) 
{
        vec_t delta(out_dim());
        const activation::function& h = layers_.tail()->activation_function();
        if (is_canonical_link(h)) 
	{
		if (typeid(h) == typeid(activation::softmax))
			for_i(out_dim(), [&](int i) { delta[i] = target_value_max(); });
		else
			for_i(out_dim(), [&](int i){ delta[i] = target_value_max() * h.df(out[i]);});
        } 
	else 
	{
            for_i(out_dim(), [&](int i){ delta[i] = target_value_max() * h.df(out[i]) * h.df(out[i]);}); // FIXME
        }
layers_.tail()->back_propagation_2nd(delta);
}

**Comment by zeno40** [comment]
example that works very good with above fix:

`network<cross_entropy_multiclass, gradient_descent_levenberg_marquardt> nn;`

nn << convolutional_layer<elu>(32, 32, 3, 3, 3, 16, padding::same)
		<< convolutional_layer<elu>(32, 32, 3, 3, 16, 16, padding::same)
		<< max_pooling_layer<activation::identity>(32, 32, 16, 2, 2)
		<< convolutional_layer<elu>(16, 16, 3, 3, 16, 24, padding::same)
		<< convolutional_layer<elu>(16, 16, 3, 3, 24, 24, padding::same)
		<< max_pooling_layer<activation::identity>(16, 16, 24, 2, 2)
		<< convolutional_layer<elu>(8, 8, 3, 3, 24, 32, padding::same)
		<< convolutional_layer<elu>(8, 8, 3, 3, 32, 32, padding::same)
		<< max_pooling_layer<activation::identity>(8, 8, 32, 2, 2)
		<< fully_connected_layer<activation::elu>(512, 10)
	    << fully_connected_layer<softmax>(10, 10);
    std::cout << "load models..." << std::endl;
	
	nn.weight_init(weight_init::he());
	nn.bias_init(weight_init::constant(0.1));
	int minibatch_size = 1;
	nn.optimizer().alpha = 0.00025;

    // load CIFAR-10 dataset
    std::vector<label_t> train_labels, test_labels;
    std::vector<vec_t> train_images, test_images;
	
	parse_cifar10(data_dir_path + "/data_batch_1.bin", &train_images, &train_labels, -1.0, 1.0, 0, 0);
	parse_cifar10(data_dir_path + "/data_batch_2.bin", &train_images, &train_labels, -1.0, 1.0, 0, 0);
	parse_cifar10(data_dir_path + "/data_batch_3.bin", &train_images, &train_labels, -1.0, 1.0, 0, 0);
	parse_cifar10(data_dir_path + "/data_batch_4.bin", &train_images, &train_labels, -1.0, 1.0, 0, 0);
	parse_cifar10(data_dir_path + "/data_batch_5.bin", &train_images, &train_labels, -1.0, 1.0, 0, 0);
	parse_cifar10(data_dir_path + "/test_batch.bin", &test_images, &test_labels, -1.0, 1.0, 0, 0);

    std::cout << "start learning" << std::endl;

**Comment by edgarriba** [comment]
@zeno40 Is this issue still happening in last version?

**Comment by zeno40** [comment]
No, gradient_descent_levenberg_marquardt is removed as optimzer

**Comment by edgarriba** [comment]
thx !
